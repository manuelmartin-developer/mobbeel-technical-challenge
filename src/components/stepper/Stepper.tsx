import React from "react";
import styles from "./Stepper.module.scss";
import { useStepperStore } from "../../store/stepper.store";

interface StepperProps {
  /**
   * The children to render inside the stepper
   * @type {React.ReactNode}
   * @memberof StepperProps
   * @required
   * @example
   * <Stepper>
   * <Step title="Welcome" description="Welcome to the document scanner app" />
   * <Step title="Front document" description="Please take a picture of the front of your document" />
   * </Stepper>
   * */
  children: React.ReactNode;
}

const Stepper: React.FC<StepperProps> = ({ children }) => {
  // Store states
  const { activeStep } = useStepperStore();
  return (
    <section className={styles.stepper}>
      <div className={styles.stepper__line}></div>
      {React.Children.map(children, (child, index) => {
        return React.cloneElement(child as React.ReactElement, {
          className: `${styles.stepper__step} ${
            index === activeStep ? styles.stepper__step_active : ""
          } ${index < activeStep ? styles.stepper__step_done : ""} ${
            index > activeStep ? styles.stepper__step_disabled : ""
          }`,
        });
      })}
    </section>
  );
};

export default Stepper;
