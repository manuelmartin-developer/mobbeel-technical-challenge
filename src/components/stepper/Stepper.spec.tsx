import "@testing-library/jest-dom";
import { describe, expect, it } from "vitest";
import { render, screen } from "@testing-library/react";

import Stepper from "./Stepper";

describe("Stepper", () => {
  it("renders Stepper component", () => {
    render(
      <Stepper>
        <div>Step 1</div>
        <div>Step 2</div>
      </Stepper>,
    );
    expect(screen.getByTestId("stepper")).toBeInTheDocument();
  });

  it("renders children", () => {
    render(
      <Stepper>
        <div>Step 1</div>
        <div>Step 2</div>
      </Stepper>,
    );
    expect(screen.getByText("Step 1")).toBeInTheDocument();
    expect(screen.getByText("Step 2")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <Stepper>
        <div>Step 1</div>
        <div>Step 2</div>
      </Stepper>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
