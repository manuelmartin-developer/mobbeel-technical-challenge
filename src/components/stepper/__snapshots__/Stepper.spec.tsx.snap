// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Stepper > matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="_stepper_464a67"
    data-testid="stepper"
  >
    <div
      class="_stepper__line_464a67"
    />
    <div
      class="_stepper__step_464a67 _stepper__step_active_464a67  "
    >
      Step 1
    </div>
    <div
      class="_stepper__step_464a67   _stepper__step_disabled_464a67"
    >
      Step 2
    </div>
  </section>
</DocumentFragment>
`;
